File: frontend/magic-and-phoenix/src/app/dto/game-dto.ts
export class GameDTO {
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get masterId(): number {
        return this._masterId;
    }
    public set masterId(value: number) {
        this._masterId = value;
    }
    public get token(): string {
        return this._token;
    }
    public set token(value: string) {
        this._token = value;
    }
    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }


    constructor(
        private _id: number,
        private _name: string,
        private _masterId: number,
        private _token: string
    ){

    }
}

File: backend/game-service/src/main/java/com/magicandphoenix/gameservice/controller/dto/GameDTO.java
package com.magicandphoenix.gameservice.controller.dto;

import com.magicandphoenix.gameservice.model.Game;

import javax.validation.constraints.NotEmpty;

public class GameDTO {

    private Long id;
    @NotEmpty
    private String name;
    @NotEmpty
    private Long masterId;
    @NotEmpty
    private String token;

    public GameDTO() {
    }

    public GameDTO(@NotEmpty String name, @NotEmpty Long masterId, @NotEmpty String token) {
        setName(name);
        setMasterId(masterId);
        setToken(token);
    }

    public GameDTO(Game game) {
        setId(game.getId());
        setName(game.getName());
        setMasterId(game.getMasterId());
        setToken(game.getToken());
    }

    public String getName() {
        return name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Long getMasterId() {
        return masterId;
    }

    public void setMasterId(Long masterId) {
        this.masterId = masterId;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }
}
